src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1092,14 +1092,19 @@ namespace cryptonote
 
       results[i].blob_size = it->blob.size();
       results[i].weight = results[i].tx.pruned ? get_pruned_transaction_weight(results[i].tx) : get_transaction_weight(results[i].tx, it->blob.size());
-      ok &= add_new_tx(results[i].tx, results[i].hash, tx_blobs[i].blob, results[i].weight, tvc[i], tx_relay, relayed);
+      if (!add_new_tx(results[i].tx, results[i].hash, tx_blobs[i].blob, results[i].weight, tvc[i], tx_relay, relayed))
+      {
+        // Transactions with big tx_extra are still ok to be mined, so just ignore them and don't return false
+        if (!tvc[i].m_tx_extra_too_big)
+          ok = false;
+      }
 
       if(tvc[i].m_verifivation_failed)
       {MERROR_VER("Transaction verification failed: " << results[i].hash);}
       else if(tvc[i].m_verifivation_impossible)
       {MERROR_VER("Transaction verification impossible: " << results[i].hash);}
 
-      if(tvc[i].m_added_to_pool)
+      if(tvc[i].m_added_to_pool && (results[i].tx.extra.size() <= MAX_TX_EXTRA_SIZE))
```

Comment:
Why is this check necessary? `m_added_to_pool` is only ever set *after* the `tx.extra` checks in `add_new_tx`. In other words, it seems redundant because `m_added_to_pool` will be false if `tx.extra` is too big anyways.

Reply:
It can happen if `kept_by_block` is true, and then I get tx spam in p2pool logs - all big tx_extra transactions get added to a new block and leak into ZMQ notifications, but they shouldn't because ZMQ notification are supposed to notify abou mempool only.

Reply:
@jeffro256 I just did a quick test by adding some logging: https://paste.debian.net/hidden/15de343a/
I can see both places filtering transactions from being sent to ZMQ subscribers, so it's correct. `tvc[i].m_added_to_pool` can be true but we still end up in the `else` branch, and the debug logging shows it.

Reply:
Correction: https://paste.debian.net/hidden/d345d788/ - now I see only the first place filtering transactions, I'll keep running it to see if the second one triggers or not.

Reply:
When a new block arrives, monerod downloads missing transactions from this block, adds them to mempool for a split second and then removes them from mempool and adds them to the blockchain. All this juggling triggers ZMQ notifications which are useless to subscribers.

Reply:
Since downloaded transactions are "forced into" the mempool  by `kept_by_block`, can we not generalize this condition to `tvc[i].m_added_to_pool && tx_relay != relay_method::block`? This would exclude any internal blockchain shufflings inside the pool, not just for `tx_extra`.

Reply:
I'll run with debug logging to confirm that it's only `relay_method::block` there, and if it is, I'll change to `&& tx_relay != relay_method::block`

Reply:
@jeffro256 I just realized that `kept_by_block` can make transactions stay in mempool for real, and be actually valid to be mined again, for example after a reorg. So it's a bad idea to change to `tx_relay != relay_method::block`, it's better to filter only big tx_extra.